File: core/src/net/jorhlok/oops/Postage.java
package net.jorhlok.oops;

import java.util.List;

/**
 * Some sort of message between entities. Don't put things that must be disposed here.
 * @author Jorhlok
 */
public class Postage {
    public Entity ReturnAddress;
    public String Name;
    public String Type;
    public int iValue;
    public float fValue;
    public String sValue;
    public List<Object> Contents; //if the above doesn't suffice
    
    public Postage() {
        ReturnAddress = null;
        Name = null;
        Type = null;
        iValue = 0;
        fValue = 0;
        sValue = null;
        Contents = null;
    }
    
    public Postage(Entity e, String name, String type) {
        ReturnAddress = e;
        Name = name;
        Type = type;
        iValue = 0;
        fValue = 0;
        sValue = null;
        Contents = null;
    }
    
    public Postage(Entity e, String name, String type, int i) {
        ReturnAddress = e;
        Name = name;
        Type = type;
        iValue = i;
        fValue = 0;
        sValue = null;
        Contents = null;
    }
    
    public Postage(Entity e, String name, String type, int i, float f, String s) {
        ReturnAddress = e;
        Name = name;
        Type = type;
        iValue = i;
        fValue = f;
        sValue = s;
        Contents = null;
    }
    
    public Postage(Entity e, String name, String type, int i, float f, String s, List<Object> c) {
        ReturnAddress = e;
        Name = name;
        Type = type;
        iValue = i;
        fValue = f;
        sValue = s;
        Contents = c;
    }
    
    public Postage(Entity e, String name, String type, List<Object> c) {
        ReturnAddress = e;
        Name = name;
        Type = type;
        iValue = 0;
        fValue = 0;
        sValue = null;
        Contents = c;
    }
}

File: core/src/net/jorhlok/oops/Postage.kt
package net.jorhlok.oops

/**
 * Some sort of message between entities. Don't put things that must be disposed here.
 * @author Jorhlok
 */
class Postage {
    var ReturnAddress: Entity? = null
    var Name: String = ""
    var Type: String = ""
    var iValue: Int = 0
    var fValue: Float = 0f
    var sValue: String = ""
    var Contents: List<Any>? = null //if the above doesn't suffice


    constructor(e: Entity?, name: String, type: String = "", i: Int = 0, f: Float = 0f, s: String = "", c: List<Any>? = null) {
        ReturnAddress = e
        Name = name
        Type = type
        iValue = i
        fValue = f
        sValue = s
        Contents = c
    }

    constructor(e: Entity?, name: String, type: String, c: List<Any>) {
        ReturnAddress = e
        Name = name
        Type = type
        Contents = c
    }
}
